conans/client/rest/rest_client.py

```diff
@@ -118,9 +119,10 @@ def get_conanfile(self, conan_reference):
 
         # TODO: Get fist an snapshot and compare files and download only required?
 
-        # Download the resources
-        contents = self.download_files(urls, self._output)
-        return contents
+        # Download the resources to a temp folder
+        temp_folder = tempfile.mkdtemp(suffix='conan_download')
```

Comment:
I'd prefer if the temp folder were located in the conan local cache storage. Probably try to completely remove it after usage (maybe not after each download, but after an install).